grpc/src/main/java/com/linecorp/armeria/server/grpc/UnframedGrpcErrorHandlerBuilder.java

```diff
@@ -0,0 +1,116 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.grpc;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.EnumSet;
+import java.util.Set;
+
+import org.curioswitch.common.protobuf.json.MessageMarshaller;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.protobuf.Message;
+
+import com.linecorp.armeria.common.annotation.Nullable;
+
+public final class UnframedGrpcErrorHandlerBuilder {
+    private UnframedGrpcStatusMappingFunction statusMappingFunction;
+
+    private MessageMarshaller jsonMarshaller = UnframedGrpcErrorHandlers.ERROR_DETAILS_MARSHALLER;
+
+    @Nullable
+    private Set<UnframedGrpcErrorResponseType> responseTypes;
+
+    UnframedGrpcErrorHandlerBuilder(UnframedGrpcStatusMappingFunction statusMappingFunction) {
+        this.statusMappingFunction = statusMappingFunction;
+    }
+
+    public UnframedGrpcErrorHandlerBuilder jsonMarshaller(MessageMarshaller jsonMarshaller) {
+        this.jsonMarshaller = requireNonNull(jsonMarshaller, "jsonMarshaller");
+        return this;
+    }
+
+    public UnframedGrpcErrorHandlerBuilder statusMappingFunction(
+            UnframedGrpcStatusMappingFunction statusMappingFunction) {
+        this.statusMappingFunction = requireNonNull(statusMappingFunction, "statusMappingFunction");
+        return this;
+    }
+
+    public UnframedGrpcErrorHandlerBuilder responseTypes(UnframedGrpcErrorResponseType... responseTypes) {
+        requireNonNull(responseTypes, "responseTypes");
+        if (this.responseTypes == null) {
+            this.responseTypes = EnumSet.noneOf(UnframedGrpcErrorResponseType.class);
+        }
+        this.responseTypes.addAll(ImmutableSet.copyOf(responseTypes));
+        return this;
+    }
+
+    public UnframedGrpcErrorHandlerBuilder registerMarshallers(Message... messages) {
+        requireNonNull(messages, "messages");
+        MessageMarshaller.Builder jsonMarshallerBuilder = jsonMarshaller.toBuilder();
+
+        for (Message message : messages) {
+            jsonMarshallerBuilder = jsonMarshallerBuilder.register(message);
+        }
+
+        jsonMarshaller = jsonMarshallerBuilder.build();
+        return this;
+    }
+
+    @SafeVarargs
+    public final UnframedGrpcErrorHandlerBuilder registerMarshallers(Class<? extends Message>... messageTypes) {
+        requireNonNull(messageTypes, "messageTypes");
+        MessageMarshaller.Builder jsonMarshallerBuilder = jsonMarshaller.toBuilder();
+
+        for (Class<? extends Message> messageType : messageTypes) {
+            jsonMarshallerBuilder = jsonMarshallerBuilder.register(messageType);
+        }
+
+        jsonMarshaller = jsonMarshallerBuilder.build();
+        return this;
+    }
+
+    public UnframedGrpcErrorHandlerBuilder registerMarshallers(Iterable<?> messagesOrMessageTypes) {
```

Comment:
Should we rename these methods to `registerJsonMarshallers()`?

Reply:
I think `registerMarshallers` might be confusing because this method register `Message` although the name doesn't contain a word "message". So renamed `registerMarshalledMessages`.